### PR TITLE
delete popup

### PR DESCRIPTION
**Pull request by jolpazy**
Popup doesn't work with mobx-react-form functions if that is ever needed.
Works fine with delete requests and any other function passed though.

### CODE REVIEW

File: web/src/main/frontend/src/base/stores/PopupStore.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import {observable, action} from 'mobx';
+
+class PopupStore {
+  @observable isPopupActive = false;
+
+  @observable yesFunction;
```
Comment (iscekic) [inline web/src/main/frontend/src/base/stores/PopupStore.js]: What is a `yesFunction`?

File: web/src/main/frontend/src/base/stores/PopupStore.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import {observable, action} from 'mobx';
+
+class PopupStore {
+  @observable isPopupActive = false;
```
Comment (iscekic) [inline web/src/main/frontend/src/base/stores/PopupStore.js]: Rename to `active`.

File: web/src/main/frontend/src/base/stores/PopupStore.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+  @observable yesFunction;
+
+  @action.bound
+  runThroughPopup = func => () => {
```
Comment (iscekic) [inline web/src/main/frontend/src/base/stores/PopupStore.js]: What does `runThroughPopup` mean? 😄

File: web/src/main/frontend/src/base/stores/PopupStore.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    this.isPopupActive = true;
+    this.yesFunction = func;
+  };
+
+  @action.bound
+  deactivatePopup() {
```
Comment (iscekic) [inline web/src/main/frontend/src/base/stores/PopupStore.js]: Rename to `hide`.

File: web/src/main/frontend/src/base/stores/PopupStore.js (hunk 1 above)
Comment (iscekic) [reply web/src/main/frontend/src/base/stores/PopupStore.js]: I guess this should be `onSuccess` or similar.

File: web/src/main/frontend/src/base/stores/PopupStore.js (hunk 3 above)
Comment (iscekic) [reply web/src/main/frontend/src/base/stores/PopupStore.js]: Should probably be named `showPopup = onConfirm => () => ...`.

File: web/src/main/frontend/src/components/Category.js [hunk 5]
```diff
@@ -24,78 +24,82 @@ class Category extends React.Component {
...
+                    />
+                    <p className="form__error_container">
+                      {categoryForm.$('description').error}
+                    </p>
+                  </div>
+                  <div className="col-12">
+                    <div className="d-flex justify-content-end">
+                      {categoryForm.$('id').values() > 0 && (
+                        <Button
+                          click={this.props.popup.runThroughPopup(
```
Comment (iscekic) [inline web/src/main/frontend/src/components/Category.js]: No need to create a click handler each render.

File: web/src/main/frontend/src/base/stores/PopupStore.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import {observable, action} from 'mobx';
+
+class PopupStore {
```
Comment (iscekic) [inline web/src/main/frontend/src/base/stores/PopupStore.js]: Should probably be named `ConfirmPopupStore`.

File: web/src/main/frontend/src/components/Formula.js [hunk 7]
```diff
@@ -26,95 +26,99 @@ class Formula extends React.Component {
...
+                    />
+                  </div>
+                  <div className="col-12 add__button_container">
+                    <Button cssClass="icon__block" title="+" />
+                  </div>
+                  <div className="col-12">
+                    <div className="d-flex justify-content-end">
+                      {formulaForm.$('id').values() > 0 && (
+                        <Button
+                          click={this.props.popup.runThroughPopup(
```
Comment (iscekic) [inline web/src/main/frontend/src/components/Formula.js]: Same here.

File: web/src/main/frontend/src/components/ui/Popup.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import Button from './Button';
+import {observer, inject} from 'mobx-react';
+
+@inject('popup')
+@observer
+class Popup extends React.Component {
```
Comment (iscekic) [inline web/src/main/frontend/src/components/ui/Popup.js]: Rename to `ConfirmPopup`.

File: web/src/main/frontend/src/components/ui/Popup.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  componentDidMount() {
+    window.addEventListener('keydown', this.handleKey);
```
Comment (iscekic) [inline web/src/main/frontend/src/components/ui/Popup.js]: Does this have to be on `window`? Can't it be on a `ref` of this component?

File: web/src/main/frontend/src/components/ui/Popup.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+  componentWillUnmount() {
+    window.removeEventListener('keydown', this.handleKey);
+  }
+
+  handleKey = e => {
+    if (e.key === 'Escape') this.props.popup.deactivatePopup();
+  };
+
+  onYes = () => {
```
Comment (iscekic) [inline web/src/main/frontend/src/components/ui/Popup.js]: `onConfirm`

File: web/src/main/frontend/src/scss/base/_base.scss [hunk 11]
```diff
@@ -54,6 +54,30 @@ textarea {
...
 }
 
+.popup-background {
+  background-color: rgba(204, 211, 224, 0.5);
+  position: fixed;
+  width: 100%;
+  height: 100%;
+  top: 0;
+  left: 0;
+  z-index: 10;
```
Comment (iscekic) [inline web/src/main/frontend/src/scss/base/_base.scss]: Double check that `z-index` values are fine.

File: web/src/main/frontend/src/base/stores/PopupStore.js (hunk 3 above)
Comment (jolpazy) [reply web/src/main/frontend/src/base/stores/PopupStore.js]: runThroughPopup is set as click handler if we want to have yes or no dialog before actually running the function (which is passed as it's argument);
Currently accepting new name propositions @vrackovic @iscekic ((:
Component is reusable in that way, but id doesn't work with submitting forms.

File: web/src/main/frontend/src/components/Category.js (hunk 5 above)
Comment (jolpazy) [reply web/src/main/frontend/src/components/Category.js]: only the value of second function changes whenever selected value inside table changes.
It basicaly says: run this function (argument) after (if) popup is confirmed

File: web/src/main/frontend/src/scss/base/_base.scss (hunk 11 above)
Comment (jolpazy) [reply web/src/main/frontend/src/scss/base/_base.scss]: They are fine now, will increase to leave some space, as popup should always get on top of everything.

File: web/src/main/frontend/src/base/stores/PopupStore.js (hunk 3 above)
Comment (jolpazy) [reply web/src/main/frontend/src/base/stores/PopupStore.js]: Ok, it's showPopup now (:

File: web/src/main/frontend/src/components/ui/Popup.js (hunk 9 above)
Comment (jolpazy) [reply web/src/main/frontend/src/components/ui/Popup.js]: Can I set "keydown" event on a div?
I want to remove popup if escape is pressed anywhere.

File: web/src/main/frontend/src/index.js [hunk 12]
```diff
@@ -31,6 +32,7 @@ class Root extends React.Component {
           completedTest={CompletedTestStore}
           newTest={NewTestStore}
           loader={LoaderStore}
+          popup={ConfirmPopupStore}
```
Comment (iscekic) [inline web/src/main/frontend/src/index.js]: `confirmPopup`

**Pull request merged**